### PR TITLE
patch

### PR DESCRIPTION
**Pull request by tannergooding**

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -1,2 +1,8 @@
 # Auto detect text files and perform LF normalization
 * text=auto
+
+###############################################################################
+# Set explicit file behavior to:
+#   treat as binary
+###############################################################################
+*.snk       binary
```
Comment (tannergooding) [inline .gitattributes:8]: It's good to be explicit with file types, otherwise they can be mispredicted

File: Directory.Build.props [hunk 2]
```diff
@@ -51,16 +51,16 @@
     <Copyright>Copyright � John Kelly and Contributors</Copyright>
     <Description>A high speed vector maths library</Description>
     <Features>strict</Features>
-    <!--<GenerateDocumentationFile>true</GenerateDocumentationFile>-->
-    <AssemblyOriginatorKeyFile>$(MSBuildThisFileDirectory)MathSharp.snk</AssemblyOriginatorKeyFile>
+    <GenerateDocumentationFile>false</GenerateDocumentationFile>
     <HighEntropyVA>true</HighEntropyVA>
     <LangVersion>preview</LangVersion>
+    <MathSharpPublicKey>00240000048000009400000006020000002400005253413100040000010001000586cd22205c2991de00dea27c3455744874d5b9c4532ccee2bf8fa9b55752af42fa6170f641c20d1501cd3b9e6d22a1327cdaf19eb8eba23e8ed350a60930e7286417916e3a3a39798e69d8d521ac3af4c720cbcca2f2fbf5cee2c0b1c6be08ebba9051399148c663c3408513be9a2ff96e7816bb1b3713d1f30e13fa2e66a9</MathSharpPublicKey>
```
Comment (tannergooding) [inline Directory.Build.props:57]: It's generally good to have your own SNK file as well. It isn't private, but it is used for "identity".

The one from TerraFX is meant for TerraFX 😛. I generated a new one for you as part of this PR.

Comment (tannergooding) [reply Directory.Build.props:57]: (you can also generate your own using the VS UX, when needed)

**Comment by john-h-k** [comment]
Thank you very much, that's v. useful

File: Directory.Build.props [hunk 3]
```diff
@@ -51,16 +51,16 @@
...
     <NeutralLanguage>en-GB</NeutralLanguage>
     <OverwriteReadOnlyFiles>true</OverwriteReadOnlyFiles>
     <PackageLicenseExpression>MIT</PackageLicenseExpression>
     <PackageProjectUrl>$(RepositoryUrl)</PackageProjectUrl>
     <ProduceReferenceAssembly>true</ProduceReferenceAssembly>
-    <PublicSign>true</PublicSign>
+    <PublicSign>false</PublicSign>
```
Comment (tannergooding) [inline Directory.Build.props:63]: No need to public sign if you have the full key.

File: samples/MathSharp.Interactive/MathSharp.Interactive.csproj [hunk 4]
```diff
@@ -4,12 +4,6 @@
     <NoWarn>1573;1591;$(NoWarn)</NoWarn>
     <OutputType>Exe</OutputType>
     <TargetFramework>netcoreapp3.0</TargetFramework>
-    <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
```
Comment (tannergooding) [inline samples/MathSharp.Interactive/MathSharp.Interactive.csproj:7]: All of this was duplicated or unnecessary. It is either handled by default or already defined in the root Directory.Build.props

File: sources/Directory.Build.props [hunk 5]
```diff
@@ -18,7 +18,7 @@
   <Import Project="$(MSBuildThisFileDirectory)..\Directory.Build.props" />
 
   <ItemGroup>
-    <InternalsVisibleTo Include="$(MSBuildProjectName).UnitTests" />
+    <InternalsVisibleTo Include="$(MSBuildProjectName).UnitTests" PublicKey="$(MathSharpPublicKey)" />
```
Comment (tannergooding) [inline sources/Directory.Build.props:21]: I haven't made this change on my side yet, but it makes it a bit easier to define IVTs for external components, such as if you are doing moqs/etc

File: tests/MathSharp.Tests/MathSharp.Tests.csproj [hunk 6]
```diff
@@ -1,25 +0,0 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (tannergooding) [inline tests/MathSharp.Tests/MathSharp.Tests.csproj:1]: `.UnitTests` is what is commonly checked for by various logic

**Pull request merged**